### PR TITLE
add simplifed  chinese locale, translation is complete now.

### PR DESCRIPTION
**Pull request by Jane620**
only for the code part and a small portion of text testing.

### CODE REVIEW

**Comment by Adri1** [comment]
I think it would be better to precise if it is traidtional Chinese or simplifed Chinese, including in the name of the folde not just named "cn"

**Comment by Aizzzs** [comment]
Hello! I can offer assistance with localizing games into Chinese. I have experience in localizing Pokémon fan games!

**Comment by TempsRay** [comment]
> I think it would be better to precise if it is traditional Chinese or simplifed Chinese, including not just naming it "cn" both in the name of the folder and in the code

I am not 100% on this but the locales for simplified Chinese should be zh-cn while Traditional should be zh-tw

**Comment by justhil** [comment]
#592 I localized the skill names and traits into Chinese.

**Review by justhil — APPROVED** [review]
frontsize is too big

**Review by justhil — APPROVED** [review]

**Review by justhil — APPROVED** [review]

File: src/system/settings.ts
[the remark this answers is not in the document]
Comment (Jane620) [reply src/system/settings.ts:1]: change 'SETTINGS_SELECTED' to 66px seems to look better, however all the icons are misaligned.

[the remark this answers is not in the document]
Comment (justhil) [reply src/system/settings.ts:1]: 对齐要我命感觉这样也不是不能玩

File: src/locales/chs/battle.ts
Comment (justhil) [inline src/locales/chs/battle.ts:1]: "pokemon": "宝可梦上吧！",  多打了个"上吧!"

File: src/locales/chs/battle.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { SimpleTranslationEntries } from "#app/plugins/i18n";
+
+export const battle: SimpleTranslationEntries = {
+  "bossAppeared": "{{bossName}} 出现了。",
+  "trainerAppeared": "{{trainerName}}\nwould like to battle!",
```
Comment (Madmadness65) [inline src/locales/chs/battle.ts:5]: It looks this one line was overlooked

File: src/system/settings.ts [hunk 2]
```diff
@@ -198,6 +198,10 @@ export function setSetting(scene: BattleScene, setting: Setting, value: integer)
                 label: 'German',
                 handler: () => changeLocaleHandler('de')
               },
+              {
+                label: 'Simplifed  Chinese',
```
Comment (Madmadness65) [inline src/system/settings.ts:202]: Minor spelling mistake, should be "Simplified"

**Review by Madmadness65 — CHANGES_REQUESTED** [review]
Just a couple small fixes requested, then it should be ready to go

**Comment by TempsRay** [comment]
> > I think it would be better to precise if it is traditional Chinese or simplifed Chinese, including not just naming it "cn" both in the name of the folder and in the code
> 
> I am not 100% on this but the locales for simplified Chinese should be zh-cn while Traditional should be zh-tw

^ Thoughts on this?

**Comment by Adri1** [comment]
`src/system/settings.ts` have just been modified with the names of the languages labeled in their repsective languages and not just in English, so you may to edit it

**Comment by Haleclipse** [comment]
1.Normalized category fixes for supportedLngs lists

Currently the language codes in "supportedLngs" are all abbreviated, which leads to an error in the automatic language recognition, at least for zh-CN recognition, which does not fall back to zh or so-called chs correctly.

2. Replace similar pixel fonts to display GBK characters correctly.

The pokemon-emerald-pro.ttf does not contain GBK characters, causing Phaser to process GBK characters with the font being displayed on the user's side. For this reason, I replaced pokemon-emerald-pro.ttf with [MuZaiPixel](https://github.com/DWNfonts/MuzaiPixel/blob/main/MZPXorig.ttf)

3. Fixed the vast majority of display overflow errors in translations due to inappropriate text lengths.

**Review by Madmadness65 — APPROVED** [review]

**Comment by davidyan-hub** [comment]
I did not have envirement to do this...
Who can modify the text.ts to change the font-family and font-size for chinese.

Code as below:

fontFamily: localStorage.getItem('prLang')==="zh_CN"?'simhei':'emerald',

styleOptions.fontFamily = localStorage.getItem('prLang')==="zh_CN"?'MuZaiPixel':'pkmnems';

and same judgement logic for changing font-size, let the size is smaller.

**Pull request merged**